python/apsis/program/procstar/agent.py

```
@@ -517,31 +516,35 @@ def more_output():
-                        0 if fd_data is None else fd_data.interval.stop,
-                        None
+                try:
+                    await self.proc.request_fd_data(
+                        "stdout",
+                        interval=Interval(
+                            0 if fd_data is None else fd_data.interval.stop,
+                            None
+                        )
                     )
-                )
-                # Wait for it.
-                async for update in self.proc.updates:
-                    match update:
-                        case FdData():
-                            fd_data = _combine_fd_data(fd_data, update)
-                            # Confirm that we've accumulated all the output as
-                            # specified in the result.
-                            assert fd_data.interval.start == 0
-                            assert fd_data.interval.stop == res.fds.stdout.length
-                            break
-
-                        case _:
-                            log.debug("expected final FdData")
+                    # Wait for it.
+                    async for update in self.proc.updates:
+                        match update:
+                            case FdData():
+                                fd_data = _combine_fd_data(fd_data, update)
+                                # Confirm that we've accumulated all the output as
+                                # specified in the result.
+                                assert fd_data.interval.start == 0
+                                assert fd_data.interval.stop == res.fds.stdout.length
+                                break
+
+                            case _:
+                                log.debug("expected final FdData")
+                except AgentMessageError as exc:
+                    if "no process" in str(exc):
+                        log.info(f"{self.run_id}: could not get final output; process already gone")
```

Do you want to swallow all agent message errors, or just the no process ones?